internal/providers/github/common.go

```diff
@@ -176,6 +181,10 @@ func (c *GitHub) ListPackagesByRepository(
 		var resp *github.Response
 		var err error
 
+		if c.packageListingClient == nil {
```

Comment:
I think the condition you're introducing on line 170 makes this code unreachable.

Reply:
thanks, I started fixing the code where I saw the crash happening and then didn't remove this condition.